cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -134,7 +134,8 @@ public MetricsTable getOrCreateResolutionMetricsTable(String v2TableName, String
       MetricsTable v3Table = DatasetsUtil.getOrCreateDataset(dsFramework, v3TableId, MetricsTable.class.getName(),
                                                              props.build(), null);
 
-      if (v2Table != null) {
+      // TODO: By default do not allow reading from v2 tables. Remove this check once CDAP-12306 is fixed
+      if (v2Table != null && cConf.getBoolean(Constants.Metrics.METRICS_V2_TABLE_SCAN_ENABLED)) {
```

Comment:
better: if the config is turned off, don't try to instantiate the v2 dataset in line 126.